pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -20,7 +20,7 @@ const (
 )
 
 // SubnetSpec defines the desired state of Subnet.
-// +kubebuilder:validation:XValidation:rule="has(oldSelf.subnetDHCPConfig) || !has(self.subnetDHCPConfig) || !has(self.subnetDHCPConfig.mode) || self.subnetDHCPConfig.mode=='DHCPDeactivated'", message="subnetDHCPConfig cannot switch from DHCPDeactivated to other modes"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.subnetDHCPConfig)==has(self.subnetDHCPConfig) || (has(oldSelf.subnetDHCPConfig) && !has(self.subnetDHCPConfig) && (!has(oldSelf.subnetDHCPConfig.mode) || oldSelf.subnetDHCPConfig.mode=='DHCPDeactivated')) || (!has(oldSelf.subnetDHCPConfig) && has(self.subnetDHCPConfig) && (!has(self.subnetDHCPConfig.mode) || self.subnetDHCPConfig.mode=='DHCPDeactivated'))", message="subnetDHCPConfig mode can only switch between DHCPServer and DHCPRelay"
```

Comment:
we should also allow if !has(oldSelf.subnetDHCPConfig) && !has(self.subnetDHCPConfig)?

Reply:
Similarly, `has(oldSelf.subnetDHCPConfig)==has(self.subnetDHCPConfig)` allows the case for both `!has(oldSelf.subnetDHCPConfig) && !has(self.subnetDHCPConfig)` and `has(oldSelf.subnetDHCPConfig) && has(self.subnetDHCPConfig)`